awx/main/scheduler/task_manager.py

```diff
@@ -412,7 +414,7 @@ def generate_dependencies(self, undeped_tasks):
                 continue
             # TODO: Can remove task.project None check after scan-job-default-playbook is removed
             if task.project is not None and task.project.scm_update_on_launch is True:
-                latest_project_update = self.get_latest_project_update(task)
+                latest_project_update = self.get_latest_project_update(task.project_id)
```

Comment:
looks like you still need to change the method signature in some other places as well.